python/lsst/meas/algorithms/dynamicDetection.py

```diff
@@ -190,14 +209,49 @@ def detectFootprints(self, exposure, doSmooth=True, sigma=None, clearMask=True,
             # Rinse and repeat thresholding with new calculated threshold
             results = self.applyThreshold(middle, maskedImage.getBBox(), factor)
             results.prelim = prelim
+            results.background = lsst.afw.math.BackgroundList()
             if self.config.doTempLocalBackground:
                 self.applyTempLocalBackground(exposure, middle, results)
             self.finalizeFootprints(maskedImage.mask, results, sigma, factor)
 
-            if self.config.reEstimateBackground:
-                self.reEstimateBackground(maskedImage, prelim)
-
             self.clearUnwantedResults(maskedImage.mask, results)
-            self.display(exposure, results, middle)
+
+        if self.config.reEstimateBackground:
+            self.reEstimateBackground(maskedImage, results.background)
+
+        self.display(exposure, results, middle)
+
+        if self.config.doBackgroundTweak:
+            # Re-do the background tweak after any temporary backgrounds have been restored
+            bgLevel = self.calculateThreshold(exposure, seed, sigma=sigma).additive
+            self.tweakBackground(exposure, bgLevel, results.background)
 
         return results
+
+    def tweakBackground(self, exposure, bgLevel, bgList=None):
+        """Modify the background to by a constant value
+
+        Parameters
+        ----------
+        exposure : `lsst.afw.image.Exposure`
+            Exposure for which to tweak background.
+        bgLevel : `float`
+            Background level to remove
+        bgList : `lsst.afw.math.BackgroundList`, optional
+            List of backgrounds to append to.
+
+        Returns
+        -------
+        bg : `lsst.afw.math.BackgroundMI`
+            Constant background model.
+        """
+        self.log.info("Tweaking background by %f to match sky photometry", bgLevel)
+        exposure.image -= bgLevel
+        bgStats = lsst.afw.image.MaskedImageF(1, 1)
+        bgStats.set(0, 0, (bgLevel, 0, bgLevel))
+        bg = lsst.afw.math.BackgroundMI(exposure.getBBox(), bgStats)
+        bgData = (bg, lsst.afw.math.Interpolate.LINEAR, lsst.afw.math.REDUCE_INTERP_ORDER,
+                  lsst.afw.math.ApproximateControl.UNKNOWN, 0, 0, False)
```

Comment:
Sad that there isn't a simpler way to make a background object that corresponds to a constant, but I'll trust that it works and that you looked for something simpler.

Reply:
Yeah, I'd love to see all the background modelling stuff redesigned, but now's not the time.